disk.go

```diff
@@ -19,21 +20,65 @@ const (
 //
 //==================================================================
 type Disk struct {
-	node       RunnerInterface
-	MBps       int
-	reserved   int
-	lastIOdone time.Time
-	writes     int64
-	reads      int64
-	writebytes int64
-	readbytes  int64
+	node           RunnerInterface
+	MBps           int
+	reserved       int
+	lastIOdone     time.Time
+	writes         int64
+	reads          int64
+	writebytes     int64
+	readbytes      int64
+	stopHandler    bool
+	pendingwbytes  int64
+	pendingrbytes  int64
+	qMutex         sync.Mutex
 }
 
 func NewDisk(r RunnerInterface, mbps int) *Disk {
 	d := &Disk{node: r, MBps: mbps, reserved: 0, lastIOdone: Now}
 	return d
 }
 
+//
+// This method is to simulate any disk controller actions.
+// The minimal handler here simulates the writes. Wakes up
+// after every unit of time needed to write a chunk and then
+// updates the pendingwbytes which is used to derive queue depth.
+//
+func (d *Disk) handler() {
+	// time required to write one chunk.
+	sleepTime := configStorage.dskdurationDataChunk
+
+	for d.stopHandler == false {
+		// Sleep for one chunk write time
+		time.Sleep(sleepTime)
+
+		// Smulate the write of one chunk
+		d.qMutex.Lock()
+		pendingwbytes := d.pendingwbytes
+		pendingwbytes -= int64(configStorage.sizeDataChunk * 1024)
+		if pendingwbytes >= 0 {
+			d.pendingwbytes = pendingwbytes
+		}
+		d.qMutex.Unlock()
+	}
+}
+
+// Start the disk functionality.
+// Model should call this when it is
+// ready to consume chunks
+func (d *Disk) Start() {
+	d.stopHandler = false
+	go d.handler()
```

Comment:
Some protection against starting two times in a row. Use asserts, here and elsewhere.